File: app/src/main/java/wseemann/media/romote/activity/SettingsActivity.java
package wseemann.media.romote.activity;

import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceActivity;
import android.preference.PreferenceFragment;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

import com.jaku.core.JakuRequest;
import com.jaku.core.KeypressKeyValues;
import com.jaku.request.KeypressRequest;

import java.util.List;

import wseemann.media.romote.R;
import wseemann.media.romote.tasks.RequestCallback;
import wseemann.media.romote.tasks.RequestTask;
import wseemann.media.romote.utils.CommandHelper;
import wseemann.media.romote.utils.RokuRequestTypes;

/**
 * Created by wseemann on 2/20/16.
 */
public class SettingsActivity extends PreferenceActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);

        LinearLayout root = (LinearLayout)findViewById(android.R.id.list).getParent().getParent().getParent();
        Toolbar bar = (Toolbar) LayoutInflater.from(this).inflate(R.layout.settings_toolbar, root, false);
        root.addView(bar, 0); // insert at top
        bar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        //getFragmentManager().beginTransaction().replace(android.R.id.content, new Prefs1Fragment()).commit();
        // Load the preferences from an XML resource
        addPreferencesFromResource(R.xml.preferences);

        findPreference("find_remote").setOnPreferenceClickListener(
                new Preference.OnPreferenceClickListener() {
                    @Override
                    public boolean onPreferenceClick(Preference preference) {
                        performKeypress(KeypressKeyValues.FIND_REMOTE);
                        return true;
                    }
                });
    }

    private void performKeypress(KeypressKeyValues keypressKeyValue) {
        String url = CommandHelper.getDeviceURL(this);

        KeypressRequest keypressRequest = new KeypressRequest(url, keypressKeyValue.getValue());
        JakuRequest request = new JakuRequest(keypressRequest, null);

        new RequestTask(request, new RequestCallback() {
            @Override
            public void requestResult(RokuRequestTypes rokuRequestType, RequestTask.Result result) {

            }

            @Override
            public void onErrorResponse(RequestTask.Result result) {

            }
        }).execute(RokuRequestTypes.keypress);
    }

    /**
     * Populate the activity with the top-level headers.
     */
    @Override
    public void onBuildHeaders(List<Header> target) {
        //loadHeadersFromResource(R.xml.preference_headers, target);
    }

    @Override
    public boolean isValidFragment(String fragmentName) {
        return true;
    }

    /**
     * This fragment shows the preferences for the first header.
     */
    /**
     * This fragment shows the preferences for the second header.
     */
    public static class Prefs1Fragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            // Can retrieve arguments from headers XML.
            Log.i("args", "Arguments: " + getArguments());

            // Load the preferences from an XML resource
            addPreferencesFromResource(R.xml.preferences);
        }
    }
}

File: app/src/main/java/wseemann/media/romote/activity/SettingsActivity.kt
package wseemann.media.romote.activity

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import dagger.hilt.android.AndroidEntryPoint
import wseemann.media.romote.composables.SettingsScreen
import wseemann.media.romote.composables.theme.RomoteTheme
import wseemann.media.romote.utils.Constants
import wseemann.media.romote.utils.openPlayStoreListing
import wseemann.media.romote.utils.enableRomoteEdgeToEdge
import wseemann.media.romote.viewmodels.SettingsScreenViewModel
import androidx.core.net.toUri

@AndroidEntryPoint
class SettingsActivity : ComponentActivity() {

    private val settingsScreenViewModel: SettingsScreenViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        enableRomoteEdgeToEdge(this)
        super.onCreate(savedInstanceState)

        setContent {
            val uiState by settingsScreenViewModel.uiState.collectAsStateWithLifecycle()

            RomoteTheme {
                SettingsScreen(
                    uiState = uiState,
                    onEvent = settingsScreenViewModel::onHandleEvent,
                    onLicensesClick = {
                        startActivity(Intent(this, LicensesActivity::class.java))
                    },
                    onRateClick = { openPlayStoreListing() },
                    onDonateClick = {
                        startActivity(
                            Intent(Intent.ACTION_VIEW, Constants.PAYPAL_DONATION_LINK.toUri())
                        )
                    },
                    onBackClick = { finish() }
                )
            }
        }
    }
}
